### DISCUSSION TITLE
New ModBus Python or Solaredge doesn't work

### DISCUSSION THREAD

**Discussion by diegoflyer** [addiejanssen/domoticz-solaredge-modbustcp-plugin#12] [Q&A] [answered]
I have already the plugin running with pymodbus 2.4.0, I tried a fresh install and the plugin was not able to connect with inverter (pymodbus 2.5.0).
I try to change the folder dist-packages with the old installation and started to work correctly......

**Comment by divx118** [comment]
Yes thanks for the heads up.
I solved it with:
```
sudo pip3 uninstall pymodbus
sudo pip3 install pymodbus==2.4.0
```

**Comment by addiejanssen** [comment, marked as answer]
Thanks for the info.

pymodbus >= 2.5.0 apparently has "new behavior" according to the [release notes](https://github.com/nmakel/solaredge_modbus/releases/tag/0.6.5) of the solaredge_modbus library (which is used by this plugin).
I will test this and see if there is anything that needs to be done in the plugin.
